activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -847,7 +847,14 @@ def configure_connection
         # http://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_strict_all_tables
         # If the user has provided another value for sql_mode, don't replace it.
         unless variables.has_key?('sql_mode') || defaults.include?(@config[:strict])
-          variables['sql_mode'] = strict_mode? ? 'STRICT_ALL_TABLES' : ''
+          if strict_mode?
+            sql_mode = "CONCAT(@@sql_mode, ',STRICT_ALL_TABLES')"
+          else
+            sql_mode = "REPLACE(@@sql_mode, 'STRICT_TRANS_TABLES', '')"
+            sql_mode = "REPLACE(#{sql_mode}, 'STRICT_ALL_TABLES', '')"
+            sql_mode = "REPLACE(#{sql_mode}, 'TRADITIONAL', '')"
+          end
+          variables['sql_mode'] = -> { sql_mode }
```

Comment:
Treating callable values as unquoted is clever but it's not clear, particularly since we're using `rescue` to handle the condition.

We could omit this from `variables` and add to the `set` statement ourselves, like we do with encoding config.